libs/ledger/src/persistent_transaction_status_cache.cpp

```diff
@@ -0,0 +1,330 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2020 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "persistent_transaction_status_cache.hpp"
+
+#include "core/serializers/main_serializer.hpp"
+#include "logging/logging.hpp"
+
+namespace fetch {
+
+namespace serializers {
+
+template <typename D>
+struct MapSerializer<fetch::ledger::StakeUpdateEvent, D>
```

Comment:
Don't these serializers need to be in a header file?

Reply:
Nope - I thought that you would be happy with this. They are only used here, in the future if they are needed elsewhere we can add them to a common header.

The serialisers as they are, add quite a bit to all the template code - so keep them in the `.cpp` file if we can I say.

Reply:
That was not an objection - I'm merely surprised that it works, is all. I was under the impression that it is necessary for our serialisers to be in header files. Score!